Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -125,9 +125,9 @@ fileprivate extension Compiler.ByteCodeGen {
         let boundaryCheck = idx == lastIdx
         let scalar = s.unicodeScalars[idx]
         if options.isCaseInsensitive && scalar.properties.isCased {
-          builder.buildMatchScalarCaseInsensitive(scalar, boundaryCheck: boundaryCheck)
+          builder.buildMatchScalarCaseInsensitive(scalar, boundaryCheck: boundaryCheck, reverse: false)
```

Comment:
Let's add an assertion at the top of this function that we're not in reverse mode.

```swift
assert(!reverse)
```

That will help us catch times we're calling this in error